File: the_algorithms_design/chapter1/selection.c
#include <stdio.h>
#include <stdlib.h>

void selection_sort(int arr[], int len){
  int i;
  int j;
  int min;

  // The outer loop goes around N times
  // the nested loop goes around n - i - 1 times
  for(i = 0; i < len; i++){
    min = i;
    for(j = i + 1; j < len; j++){
      if(arr[j] < arr[min]){
        min = j;
      }
    }
    int temp = arr[min];
    arr[min] = arr[i];
    arr[i] = temp;
  }
}

int main(void) { 

  int arr[10] = {11,90,10,9,6,2,6,1,4,5};
  int i = 0;
  selection_sort(arr, 10);
  for(i = 0; i < 10; i++){
    printf("%d\n", arr[i]);
  }
  return 0; }

File: the_algorithms_design/chapter1/selection.py

def selec_sort(arr):
    for i in range(0,len(arr)):
        _min = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[_min]: _min = j
        arr[_min], arr[i] = arr[i], arr[_min]
    return arr


print(selec_sort([10,1,5,10,4,7,8,9]))

